### PR TITLE
Release branch 2.16.0

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :scroll: Docs          |

## Description

Blocked by #8057 #8059 and #8060. We're going to also wait some time for users to report bugs and issue in 2.16.0b0.

### CODE REVIEW

File: doc/whatsnew/2/2.16/index.rst [hunk 1]
```diff
@@ -32,3 +32,654 @@ any experience with our code!
 Last but not least @clavedeluna and @nickdrozd became triagers, welcome to the team !
 
 .. towncrier release notes start
+
+What's new in Pylint 2.16.0?
+----------------------------
+Release date: 2023-01-15
```
Comment (Pierre-Sassoulas) [inline doc/whatsnew/2/2.16/index.rst:38]:
```suggestion
Release date: 2023-01-18
```
@DanielNoord what do you think ? I will be partially unavailable from January the 24th to the 30th so either we release around that time or we could release in February.

Comment (DanielNoord) [reply doc/whatsnew/2/2.16/index.rst:38]: 18th sounds good to me!

**Review by DanielNoord — DISMISSED** [review]

**Review by DanielNoord — DISMISSED** [review]

**Comment by cdce8p** [comment]
Seems like this is the last open item for the `2.16.0` milestone https://github.com/PyCQA/pylint/milestone/71

**Comment by Pierre-Sassoulas** [comment]
I'll regenerate the changelog and release this morning, thank you for all the work Marc ! :)

**Review by cdce8p — APPROVED** [review]
LGTM! Thanks @Pierre-Sassoulas, let's do this 🚀 It's about time 😄

**Pull request merged**